### ISSUE TITLE
Unable to flash CHIP pro

### ISSUE THREAD

**Issue by JubaHub** [NextThingCo/CHIP-SDK#35]
Hi all, I have generated an ubi fs using yocto and now I want to try it on the CHIP pro board.

Following the chip-fel-flash.sh instruction that I found here [https://gitlab.ii.org.nz/iichip/meta-chip/tree/master](url) I create a shared folder in /media/sf_shared and I put an images directory with:
- roofs.ubi
- sunxi-spl.bin
- sunxi-spl-with-ecc.bin
- u-boot-dtb.bin

Using the CHIP-SDK I run
`vagrant@vagrant-ubuntu-trusty-32:~/CHIP-tools$ sudo BUILDROOT_OUTPUT_DIR=/media/sf_shared ./chip-fel-flash.sh`

The output was:
`BUILDROOT_OUTPUT_DIR = /media/sf_shared
== preparing images ==
PADDED_SPL_SIZE=0x000000c4
0+1 records in
1+0 records out
4194304 bytes (4.2 MB) copied, 0.00999624 s, 420 MB/s
UBOOT_SIZE=0x00400000
PADDED_UBOOT_SIZE=0x400000
0+0 records in
0+0 records out
0 bytes (0 B) copied, 0.000506582 s, 0.0 kB/s
echo Configuring Video Mode
reset
Image Name:   flash CHIP
Created:      Wed Apr  5 07:01:50 2017
Image Type:   ARM Linux Script (uncompressed)
Data Size:    1022 Bytes = 1.00 kB = 0.00 MB
Load Address: 00000000
Entry Point:  00000000
Contents:
   Image 0: 1014 Bytes = 0.99 kB = 0.00 MB
== upload the SPL to SRAM and execute it ==
waiting for fel.........OK
== upload spl ==
== upload u-boot ==
== upload u-boot script ==
== upload ubi ==
100% [================================================] 207618 kB,  471.0 kB/s
== execute the main u-boot binary ==
== write ubi ==
flashing...........................................................................................................................................................................................................TIMEOUT
ERROR: could not flash:`

As you can see it seems that the board never wake up from the "flashing" state.
I tryed to double the timeout time in the common.sh script at line 228.
I also tryed to pass the AFTER_FLASHING variable set to boot from the command line, but nothing changed...
From command line on my host machine I can't see the usb id of the chip pro board (0525:a4aa), there are only theese:
`Bus 002 Device 005: ID 0403:6015 Future Technology Devices International, Ltd Bridge(I2C/SPI/UART/FIFO)
Bus 002 Device 006: ID 1f3a:efe8 Onda (unverified) V972 tablet in flashing mode
Bus 002 Device 004: ID 1a40:0101 Terminus Technology Inc. Hub
`

Any idea on how to flash the CHIP pro board?

Thank you in advance,
Stefano.

**Comment by texierp** [comment]
Hello @JubaHub,

The main problem depends on your old Kernel. Try to update your layer in order to have the most recent updates.

For further informations, you can check my personal layer :  https://github.com/texierp/meta-chip (only C.H.I.P is supported for now, but you can applied this for the C.H.I.P pro (only dtb, defconfig are different)).

BR

PJ

**Comment by JubaHub** [comment]
Thank you @texierp for your answer!
I am trying to build the image with your layer, but I have the following error with u-boot...
```
ERROR: Function failed: do_install (log file is located at .../build/tmp/work/chip-poky-linux-gnueabi/u-boot-chip/2016.01+gitAUTOINC+ccd1de00d5-r0/temp/log.do_install.22786)
ERROR: Logfile of failure stored in: .../build/tmp/work/chip-poky-linux-gnueabi/u-boot-chip/2016.01+gitAUTOINC+ccd1de00d5-r0/temp/log.do_install.22786
Log data follows:
| DEBUG: Executing shell function do_install
| install: cannot create regular file '.../build/tmp/work/chip-poky-linux-gnueabi/u-boot-chip/2016.01+gitAUTOINC+ccd1de00d5-r0/image/boot/spl/sunxi-spl.bin-chip-2016.01+gitAUTOINC+ccd1de00d5-r0': No such file or directory
| WARNING: .../build/tmp/work/chip-poky-linux-gnueabi/u-boot-chip/2016.01+gitAUTOINC+ccd1de00d5-r0/temp/run.do_install.22786:1 exit 1 from
|   install .../build/tmp/work/chip-poky-linux-gnueabi/u-boot-chip/2016.01+gitAUTOINC+ccd1de00d5-r0/git/spl/sunxi-spl.bin .../build/tmp/work/chip-poky-linux-gnueabi/u-boot-chip/2016.01+gitAUTOINC+ccd1de00d5-r0/image/boot/spl/sunxi-spl.bin-chip-2016.01+gitAUTOINC+ccd1de00d5-r0
| ERROR: Function failed: do_install (log file is located at .../build/tmp/work/chip-poky-linux-gnueabi/u-boot-chip/2016.01+gitAUTOINC+ccd1de00d5-r0/temp/log.do_install.22786)
ERROR: Task 99 (.../meta-chip/recipes-bsp/u-boot/u-boot-chip_git.bb, do_install) failed with exit code '1'
```

**Comment by texierp** [comment]
Hi @JubaHub,

Very strange because it works for me.

 Can try to run the following commands : 

- `bitbake -c cleansstate u-boot-chip `
- `bitbake u-boot-chip`

PJ

**Comment by JubaHub** [comment]
@texierp I tried to rebuild the package many times...
I am building the yocto fido release. With fido your layer doesn't work (u-boot does not build) so I use the initial layer from https://github.com/myfreescalewebpage/meta-chip with your newer kernel.
The build finish, but again I am unable to flash my chip pro, with the same "timeout" error...
I also tried the flash nand image script from SDK, creating the image as in your readme of this layer, but no way, it doesn't work...

**Comment by texierp** [comment]
Hi @JubaHub,

Fido is old ... try with Krogoth or morty.

when you flashed your image you have tried this:

1. Clone SDK :
	$: git clone https://github.com/NextThingCo/CHIP-SDK.git
	$: cd CHIP-SDK
	$ ./setup_ubuntu1404.sh
2. NAND Image creation :	
	$: ./chip-create-nand-images.sh <patch to>/chip-build/tmp/work/chip-poky-linux-gnueabi/u-boot-chip/2016.01*/git <patch to>/chip-build/tmp/deploy/images/chip/<image>.tar my-chip-image 
3. Flashing (Fastboot) :
	$: sudo chown -R username:username my-chip-image/ 
	$: sudo ./chip-flash-nand-images.sh my-chip-image/ 

PJ

**Comment by JubaHub** [comment]
Yes I follow the 3 step procedure (2 and 3 are made inside the vm using vagrant user), but using a different u-boot, the 2015-10 .
Now I try to build your layer with poky master.

Should I change something in defconfig in order to run image in chip pro?

Thank you for your support,
Stefano.

**Comment by texierp** [comment]
Ok, maybe you can checkout on krogoth if you want.

Regarding the C.H.I.P, I plan to work on this week.
Some modifications are mandatory (new dts, new defconfig, ...). 

First step consist to port https://github.com/NextThingCo/CHIP-buildroot/blob/chip/stable/configs/chippro_defconfig to meta-chip.

PJ

**Comment by texierp** [comment]
Hi @JubaHub 

I've just updated my meta to add C.H.I.P Pro support : https://github.com/texierp/meta-chip/commit/f490fa0c3fd4cf5f0a16028f70dfb967f3e1d876

Not tested from my side (No C.H.I.P pro to test this)

BR

PJ

**Comment by JubaHub** [comment]
Thank you @texierp I'll test your work soon.

Stefano.
